Regression for overconstrained equilibrium on Intel CI

<!-- Please fill in the following information to report a problem with Cantera. If you have a question about using Cantera, please post it on our Google Users' Group (https://groups.google.com/forum/#!forum/cantera-users). Feature enhancements should be discussed in the dedicated Cantera enhancements repository (https://github.com/Cantera/enhancements/new/choose) -->

**Problem description**

<!-- A clear and concise description of what the bug is. -->

There's a new failure for CI runs on [Intel oneAPI](https://github.com/Cantera/cantera/actions/runs/4629308970/jobs/8189412237?pr=1411)

```
*********************************** Testing Summary ************************************
FAILED: One or more tests failed.

Tests passed: 5218
Up-to-date tests skipped: 0
scons: *** [test_results] Explicit exit, status 1
Tests failed: 1

Failed tests:
    - python: test_equilibrium.ChemEquilTest.test_equil_overconstrained2

****************************************************************************************
```

The relevant error stack is:

```
__________________ ChemEquilTest.test_equil_overconstrained2 ___________________

self = <python.test_equilibrium.ChemEquilTest testMethod=test_equil_overconstrained2>

    def test_equil_overconstrained2(self):
        gas = ct.Solution("equilibrium.yaml", "overconstrained-2")
        gas.TPX = 301, 100000, 'CH4:1.0, O2:1.0'
>       gas.equilibrate('TP', self.solver)

test/python/test_equilibrium.py:70: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   self.thermo.equilibrate(stringify(XY.upper()), stringify(solver), rtol,
E   cantera._utils.CanteraError: 
E   *******************************************************************************
E   CanteraError thrown by solve(DenseMatrix& A, double* b):
E   DGETRF returned INFO = 2. U(i,i) is exactly zero. The factorization has been completed, but the factor U is exactly singular, and division by zero will occur if it is used to solve a system of equations.
E   *******************************************************************************

build/python/cantera/thermo.pyx:438: CanteraError`
```

**Steps to reproduce**

<!-- A minimal, complete, and reproducible example demonstrating the problem. See https://stackoverflow.com/help/minimal-reproducible-example for additional suggestions on how to create such an example. -->

Occurs on any recent CI run where the intel toolchain is successfully downloaded.

**Behavior**

<!-- Describe the result of executing the above steps, and how this differs from what you expect to happen. -->

**System information**

- Cantera version: current `main`
- OS: ubuntu / GH Action `linux-intel-oneapi`
- Python/MATLAB/other software versions: Python 3.8

--- a/src/equil/ChemEquil.cpp
+++ b/src/equil/ChemEquil.cpp
@@ -258,8 +258,10 @@ int ChemEquil::estimateElementPotentials(ThermoPhase& s, vector_fp& lambda_RT,
         b[m] = mu_RT[m_component[m]];
     }
 
-    int info = solve(aa, b.data());
-    if (info) {
+    int info;
+    try {
+        info = solve(aa, b.data());
+    } catch (CanteraError&) {
         info = -2;
     }
     for (size_t m = 0; m < m_nComponents; m++) {

